src/main/java/io/kokuwa/keycloak/metrics/store/MicrometerEventStoreProviderFactory.java

```diff
@@ -0,0 +1,36 @@
+package io.kokuwa.keycloak.metrics.store;
+
+import io.kokuwa.keycloak.metrics.CommunityProfiles;
+import org.keycloak.Config;
+import org.keycloak.connections.jpa.JpaConnectionProvider;
+import org.keycloak.events.EventStoreProvider;
+import org.keycloak.events.jpa.JpaEventStoreProviderFactory;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.provider.EnvironmentDependentProviderFactory;
+
+
+public class MicrometerEventStoreProviderFactory extends JpaEventStoreProviderFactory implements EnvironmentDependentProviderFactory {
+
+    public static final String ID = "jpa2"; //override
```

Comment:
?

Reply:
Sorry. The 'jpa2' is from a failed test attempt. I've reverted it.



I've tried the same approach as for the keycloak-jpa-cache where JpaCacheDatastoreProviderFactory is overwritten based on a EnvironmentDependentProviderFactory setting. 

I would expect that if is set KC_COMMUNITY_EVENTS_METRICS_ENABLED: "true", keycloak will use the MicrometerEventStoreProviderFactory and if I set KC_COMMUNITY_EVENTS_METRICS_ENABLED: "false", keycloak would use the JpaEventStoreProviderFactory which is the default event store 

If I set KC_COMMUNITY_EVENTS_METRICS_ENABLED: "false" I get the following error. 

```
eption: Cannot invoke "org.keycloak.events.EventStoreProvider.createQuery()" because "eventStore" is null
keycloak_1  |   at org.keycloak.services.resources.admin.RealmAdminResource.getEvents(RealmAdminResource.java:789)
keycloak_1  |   at org.keycloak.services.resources.admin.RealmAdminResource$quarkusrestinvoker$getEvents_a2db18780b25698a6e5773dab7e426f2ad14dd40.invoke(Unknown Source)
keycloak_1  |   at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
keycloak_1  |   at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
keycloak_1  |   at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
keycloak_1  |   at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:582)
keycloak_1  |   at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
keycloak_1  |   at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
keycloak_1  |   at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
keycloak_1  |   at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRu
```

````
keycloak_1  | 2024-06-18 06:06:23,263 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-7) Uncaught server error: java.lang.NullPointerException: Cannot invoke "org.keycloak.events.EventStoreProvider.createAdminQuery()" because "eventStore" is null
keycloak_1  |   at org.keycloak.services.resources.admin.RealmAdminResource.getEvents(RealmAdminResource.java:875)
keycloak_1  |   at org.keycloak.services.resources.admin.RealmAdminResource$quarkusrestinvoker$getEvents_dbc7fea4b06450a9ad969f237cc951e9d7061788.invoke(Unknown Source)
keycloak_1  |   at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
keycloak_1  |   at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
keycloak_1  |   at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
keycloak_1  |   at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:582)
keycloak_1  |   at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
keycloak_1  |   at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)

````


Could you please confirm if the approach is correct?